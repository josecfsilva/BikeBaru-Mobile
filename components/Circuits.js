import React from 'react';
import { StyleSheet, FlatList, Text, View, RefreshControl, Button } from 'react-native';
import { Icon } from 'react-native-elements';

const SERVER_URL = 'https://bikebaru-server.herokuapp.com';

export default class Circuits extends React.Component {

    constructor() {
        super();
        this.state = {
            circuits: [],
        };
    }

    componentDidMount() {
        this.loadCircuits();
    }

    /* List Circuits */
    loadCircuits = () => {
        fetch(SERVER_URL + '/circuits/')
            .then(response => response.json())
            .then(response => this.setState({
                circuits: response.data,
            }))
            .catch(err => console.log(err))
    }

    /* Get Circuits */
    static getCircuits() {
        return fetch(SERVER_URL + '/circuits/')
            .then((response) => {
                return response.json();
            })

            .catch(err => console.log(err))
    }

    render() {
        return (
            <View style={styles.container}>
                <Icon name='refresh' type='foundation' onPress={this.loadCircuits} />
                <FlatList data={this.state.circuits} keyExtractor={(item, index) => index.toString()}
                    renderItem={({ item }) =>
                        <View style={{ backgroundColor: '#2f95dc', padding: 10, margin: 10, alignItems: "center" }}>
                            <Text style={{ color: '#fff', fontWeight: 'bold' }}>{item.initial_location} > {item.final_location}</Text>
                            <Text>Distância: {item.distance} km | Tempo: {item.time}</Text>
                            <Text>Velocidade Média: {item.velocity} km/h</Text>
                        </View>
                    }
                />
            </View>
        );
    }
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        justifyContent: 'center',
        alignItems: 'center',
    },
});