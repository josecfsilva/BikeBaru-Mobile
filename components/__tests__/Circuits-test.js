import React from 'react';
import 'react-native';
import Circuits from '../Circuits';
import renderer from 'react-test-renderer';
import 'isomorphic-fetch';

/* Circuit Renders Test */
test('Renders Test Case - Circuits', () => {
    const snap = renderer.create(<Circuits />).toJSON();
    expect(snap).toMatchSnapshot();
});

// API Test
it('Api Test Case - Circuits', async function () {
     global.fetch = jest.fn().mockImplementation(() => {
        var p = new Promise((resolve, reject) => {
            resolve({
                json: function () {
                    return { circuit_id: 1 }
                }
            })
        })
        return p;
    }) 
    const response = await Circuits.getCircuits();

    expect(response.circuit_id).toBe(1);
})
