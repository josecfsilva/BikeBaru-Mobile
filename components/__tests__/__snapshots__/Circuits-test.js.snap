// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Test Case - Circuits 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "justifyContent": "center",
        }
      }
    >
      <Text />
    </View>
  </View>
  <RCTScrollView
    data={Array []}
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View />
  </RCTScrollView>
</View>
`;
